app.js

```diff
@@ -24,30 +34,101 @@ const app = express()
 app.set("views", __dirname)
 app.set("view engine", "ejs")
 
+passport.use(
+  new LocalStrategy((username, password, done) => {
+    User.findOne({ username: username }, (err, user) => {
+      if (err) {
+        return done(err)
+      }
+      if (!user) {
+        return done(null, false, { message: "Incorrect username" })
+      }
+      bcrypt.compare(password, user.password, (err, result) => {
+        if (result) {
+          return done(null, user)
+        } else {
+          return done(null, false, { message: "Incorrect password" })
+        }
+      })
+    })
+  })
+)
+
+passport.serializeUser((user, done) => {
+  done(null, user.id)
+})
+
+passport.deserializeUser((id, done) => {
+  User.findById(id, (err, user) => {
+    done(err, user)
+  })
+})
 app.use(
   session({
```

Comment:
I actually appreciate you not fixing the indentation here in a certain sense (assuming you didn't write the surrounding code). In general, formatting adjustments make the diff view larger for your reviewer, and in a professional context it's always best to make the minimal change that achieves what you're trying to do. Or, separate changes with different purposes into different commits i.e. functional changes vs formatting changes. Nice work!